framework/quota/test/org/apache/cloudstack/quota/QuotaManagerImplTest.java

```diff
@@ -0,0 +1,203 @@
+
+import com.cloud.usage.UsageVO;
+import com.cloud.usage.dao.UsageDao;
+import com.cloud.user.Account;
+import com.cloud.user.AccountVO;
+import com.cloud.user.dao.AccountDao;
+import com.cloud.utils.Pair;
+import com.cloud.utils.db.TransactionLegacy;
+import junit.framework.TestCase;
+import org.apache.cloudstack.framework.config.dao.ConfigurationDao;
+import org.apache.cloudstack.quota.dao.QuotaAccountDao;
+import org.apache.cloudstack.quota.dao.QuotaBalanceDao;
+import org.apache.cloudstack.quota.dao.QuotaTariffDao;
+import org.apache.cloudstack.quota.dao.QuotaUsageDao;
+import org.apache.cloudstack.quota.dao.ServiceOfferingDao;
+import org.apache.cloudstack.quota.vo.QuotaAccountVO;
+import org.apache.cloudstack.quota.vo.QuotaTariffVO;
+import org.apache.cloudstack.quota.vo.QuotaUsageVO;
+import org.apache.cloudstack.usage.UsageTypes;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import javax.naming.ConfigurationException;
+import java.lang.reflect.Field;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RunWith(MockitoJUnitRunner.class)
+public class QuotaManagerImplTest extends TestCase {
+
+    @Mock
+    private AccountDao accountDao;
+    @Mock
+    private QuotaAccountDao quotaAcc;
+    @Mock
+    private UsageDao usageDao;
+    @Mock
+    private QuotaTariffDao quotaTariffDao;
+    @Mock
+    private QuotaUsageDao quotaUsageDao;
+    @Mock
+    private ServiceOfferingDao serviceOfferingDao;
+    @Mock
+    private QuotaBalanceDao quotaBalanceDao;
+    @Mock
+    private ConfigurationDao configDao;
+
+    @Spy
+    QuotaManagerImpl quotaManager = new QuotaManagerImpl();
+
+    private void injectMockToField(Object mock, String fieldName) throws NoSuchFieldException, IllegalAccessException {
+        Field f = QuotaManagerImpl.class.getDeclaredField(fieldName);
+        f.setAccessible(true);
+        f.set(quotaManager, mock);
+    }
+
+    @Before
+    public void setup() throws IllegalAccessException, NoSuchFieldException, ConfigurationException {
+        // Dummy transaction stack setup
+        TransactionLegacy.open("QuotaManagerImplTest");
+
+        injectMockToField(accountDao, "_accountDao");
+        injectMockToField(quotaAcc, "_quotaAcc");
+        injectMockToField(usageDao, "_usageDao");
+        injectMockToField(quotaTariffDao, "_quotaTariffDao");
+        injectMockToField(quotaUsageDao, "_quotaUsageDao");
+        injectMockToField(serviceOfferingDao, "_serviceOfferingDao");
+        injectMockToField(quotaBalanceDao, "_quotaBalanceDao");
+        injectMockToField(configDao, "_configDao");
+    }
+
+    @Test
+    public void testStartStop() {
+        try {
+            quotaManager.start(); // expected to fail as pid is not available
```

Comment:
If this is expected to fail the line 102 should call ``fail`` since it is expected that this operation will throw an exception.